tools/ci/ci_agent.py

```diff
@@ -278,6 +326,78 @@ def _run_targets(targets, src, build_path, config):
     return result
 
 
+def _build_code_coverage(src, build_path, build_flags, c_flags, codecov_token):
+    """
+    Private code coverage function.
+
+    Parameters:
+    src: location of csdk src
+    build_path: location of build dir
+    build_flags: Array of build flags to setup cmake build.
+    c_flags: Array of c flags used by compiler
+    codecov_token: codecov token required to upload code to codecov.io
+    """
+    result = {}
+    try:
+        target_result = result.setdefault("coverage", {})
+        target_build_result = target_result.setdefault("CodeCoverage", {})
+        out = _build_target("coverage", src, build_path, build_flags, c_flags)
+        log(out)
+        out = _run_cmd(f"gcovr -r . -x -o {build_path}/cobertura.xml")
```

Comment:
Looks like this might be passed to a shell - so what happens if build path contains a space?

Reply:
or worse, a semicolon...